lighthouse/tests/beacon_node.rs

```diff
@@ -0,0 +1,685 @@
+
+/// Returns the `lighthouse beacon_node --immediate-shutdown` command.
+fn base_cmd() -> Command {
+    let target_dir = env!("CARGO_BIN_EXE_lighthouse");
+    let path = target_dir
+        .parse::<PathBuf>()
+        .expect("should parse CARGO_TARGET_DIR");
+
+    let mut cmd = Command::new(path);
+    cmd.arg(BEACON_CMD)
+        .arg(format!("--{}", IMMEDIATE_SHUTDOWN_CMD));
+
+    cmd
+}
+
+/// Executes a `Command`, returning a `Result` based upon the success exit code of the command.
+fn output_result(cmd: &mut Command) -> Result<Output, String> {
+    let output = cmd.output().expect("should run command");
+
+    if output.status.success() {
+        Ok(output)
+    } else {
+        Err(from_utf8(&output.stderr)
+            .expect("stderr is not utf8")
+            .to_string())
+    }
+}
+
+// Wrapper around `Command` for easier Command Line Testing.
+struct CommandLineTest {
+    cmd: Command,
+}
+impl CommandLineTest {
+    fn new() -> CommandLineTest {
+        let base_cmd = base_cmd();
+        CommandLineTest { cmd: base_cmd }
+    }
+
+    fn flag(mut self, flag: &str, value: Option<&str>) -> Self {
+        // Build the command by adding the flag and any values.
+        self.cmd.arg(format!("--{}", flag));
+        if let Some(value) = value {
+            self.cmd.arg(value);
+        }
+        self
+    }
+
+    fn run(&mut self) -> CompletedTest {
+        // Setup temp directories.
+        let tmp_dir = TempDir::new().expect("Unable to create temporary directory");
+        let tmp_path: PathBuf = tmp_dir.path().join(CONFIG_NAME);
+
+        // Add --datadir <temp_dir> --dump-config <temp_path> to cmd.
+        self.cmd
+            .arg("--datadir")
+            .arg(tmp_dir.path().as_os_str())
+            .arg(format!("--{}", DUMP_CONFIG_CMD))
+            .arg(tmp_path.as_os_str());
+
+        // Run the command.
+        let _output = output_result(&mut self.cmd).expect("Unable to run command");
+
+        // Grab the config.
+        let config: Config =
+            from_reader(File::open(tmp_path).expect("Unable to open dumped config"))
+                .expect("Unable to deserialize to ClientConfig");
+        CompletedTest {
+            config,
+            dir: tmp_dir,
+        }
+    }
+}
+struct CompletedTest {
+    config: Config,
+    dir: TempDir,
+}
+impl CompletedTest {
+    fn with_config<F: Fn(&Config)>(self, func: F) {
+        func(&self.config);
+    }
+    fn with_config_and_dir<F: Fn(&Config, &TempDir)>(self, func: F) {
+        func(&self.config, &self.dir);
+    }
+}
+
+#[test]
+fn datadir_flag() {
+    CommandLineTest::new()
+        .run()
+        .with_config_and_dir(|config, dir| assert_eq!(config.data_dir, dir.path().join("beacon")));
+}
+
+#[test]
+fn staking_flag() {
+    CommandLineTest::new()
+        .flag("staking", None)
+        .run()
+        .with_config(|config| {
+            assert!(config.http_api.enabled);
+            assert!(config.sync_eth1_chain);
```

Comment:
Could we also check the default eth1 endpoints here too, please? It'd be great to know if they change on us.

https://github.com/sigp/lighthouse/blob/3a24ca5f14c6e6d6612fd43eca82aa0c1e6aba16/beacon_node/eth1/src/service.rs#L322